include/phasar/PhasarLLVM/DataFlowSolver/Mono/Solver/InterMonoSolver.h

```diff
@@ -344,16 +346,67 @@ template <typename AnalysisDomainTy, unsigned K> class InterMonoSolver {
     }
   }
 
-  mono_container_t
-  summarize(/*Function CalleeTarget, mono_container_t FactAtCall */) {
-    // spawn a new analysis with its own worklist
-  }
+   mono_container_t summarize(f_t CalleeTarget, mono_container_t FactAtCall) {
+    mono_container_t ResultSummary;
+   // auto EntryPoint = ICF->getFunctionName(CalleeTarget); //initialize the entrypoint to the name of CalleeTarget function
+    IntraMonoSolver<AnalysisDomainTy> IMSolver(&IMProblem); //I have problem here creating instance, I tried many ways but recieving error non-matching constructor
+   // providing the reference of IMProblem
+
+    IMSolver.solve(); //calling solve method to analyse the referred problem by IntraMonoSolver
+    
+    auto edges = ICF->getAllControlFlowEdges(CalleeTarget); //ControlflowEdges of CalleeTarget is stored in edges variable
+    for(auto &[Src, Dst]: edges){ //iterate through edges and find the exit statement
+      if(ICF->isExitStmt(Src)){  
+       ResultSummary = IMSolver.getResultsAt(Src);  //get the analysis result of Nth pr return instruction
+      }
+    }
+   return ResultSummary; //return the results
+   }
+
 
-  bool isSensibleToSummarize() {
+
+  bool isSensibleToSummarize(n_t functionNode) {
```

Comment:
`CallSite` rather than `functionNode`, right?

Reply:
changed the variable name to CallSite in commit f67db78d18d8c4d1110699e239ef4e9d945f1789